tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +218,407 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      constructor: ->
+        @_rendered = []
+
+    # Helper function to trigger a page rendering
+    renderPage = (doc, index) ->
+      if doc.isPageMapped(index)
+        throw new Error "That page is already rendered!"
```

Comment:
Perhaps something more descriptive here so you actually know what went wrong?

    "Cannot call renderPage with an already mapped index: ${index}, ensure the document is setup correctly"
    "Cannot call unrenderPage with an unmapped index: ${index}, ensure the document is setup correctly"